tutor/templates/local/docker-compose.yml

```diff
@@ -44,7 +44,7 @@ services:
       --character-set-server=utf8mb4
       --collation-server=utf8mb4_unicode_ci
       --binlog-expire-logs-seconds=259200
-      --mysql-native-password=ON
+      {% if DOCKER_IMAGE_MYSQL >= "docker.io/mysql:8.4.0" -%}--mysql-native-password=ON{%- endif %}
```

Comment:
@Danyal-Faheem I'm assuming that this condition says If MySQL has a version greater than `8.4.0` then add `--mysql-native-password=ON`.

This condition fails for the following cases on my system:
1. `tutor config save --set DOCKER_IMAGE_MYSQL=docker.io/mysql:8.14.0`
2. `tutor config save --set DOCKER_IMAGE_MYSQL=docker.io/mysql:10.0.0`

**Reason**: The >= operator is typically used for numerical or lexical (alphabetical) comparisons. Since Docker image versions are strings, this condition is comparing the strings lexicographically, not numerically.

Reply:
Great catch, I didn't think of the edge case. Thank you for picking it out. Let me update this to a number based comparison.